contrib/babelfishpg_tds/src/backend/tds/tdsprotocol.c

```diff
@@ -77,6 +79,45 @@ static void ResetTDSConnection(void);
 static TDSRequest GetTDSRequest(bool *resetProtocol);
 static void ProcessTDSRequest(TDSRequest request);
 
+
+static void attention_listener_handler(SIGNAL_ARGS)
+{
+	int	save_errno;
+
+	if (TdsRequestCtrl->phase == TDS_REQUEST_PHASE_FETCH || TdsRequestCtrl->phase == TDS_REQUEST_PHASE_INIT || TdsRequestCtrl->phase == TDS_REQUEST_PHASE_FLUSH)
+		return ;
+
+	elog(LOG, "Attention signal received for process: %d", getpid());
+
+	MemoryContext	oldContext = MemoryContextSwitchTo(TdsRequestCtrl->requestContext);
+	if (TdsReadNextBuffer() == EOF)
+		/* TODO: should be FATAL? */
+		elog(LOG, "Reading OOB data into buffer failed");
+	MemoryContextSwitchTo(oldContext);
+
+	/* TODO: If OOB request is not a TDS_ATTENTION then it is protocol violation. */
```

Comment:
What's the ideal behavior here? LOG? Any way to send a message back to the client letting them know their request is being ignored?